Scripts/BuildPhases/LintAppLocalizedStringsUsage.swift

```diff
@@ -0,0 +1,324 @@
+                referrers[childID] = objectID
+                recurseIfGroup(objectID: childID, indent: "\(indent)  ")
+            }
+        }
+        recurseIfGroup(objectID: self.pbxproj.rootProject.mainGroup)
+        return referrers
+    }()
+}
+
+// Convenience methods and properties
+extension Xcodeproj {
+    /// Builds an `Xcodeproj` instance by parsing the an `.xcodeproj` or `pbxproj` file at the provided path
+    convenience init(path: String) throws {
+        try self.init(url: URL(fileURLWithPath: path))
+    }
+
+    /// The directory where the `.xcodeproj` resides. 
+    var projectDirectory: URL { projectURL.deletingLastPathComponent().deletingLastPathComponent() }
+    /// The list of `PBXNativeTarget` targets in the project. Convenience getter for `PBXProjFile.nativeTargets`
+    var nativeTargets: [PBXNativeTarget] { pbxproj.nativeTargets }
+    /// The list of `PBXBuildFile` files a given `PBXNativeTarget` will build. Convenience getter for `PBXProjFile.buildFiles(for:)`
+    func buildFiles(for target: PBXNativeTarget) -> [PBXBuildFile] { pbxproj.buildFiles(for: target) }
+
+    /// Finds the full path / URL of a `PBXBuildFile` based on the groups it belongs to and their `sourceTree` attribute
+    func resolveURL(to buildFile: PBXBuildFile) -> URL {
+        do {
+            let fileRef = try self.pbxproj.object(id: buildFile.fileRef) as PBXFileReference
+            return resolveURL(objectUUID: buildFile.fileRef, object: fileRef)
+        } catch {
+            // Cover `XCVersionGroup` (like `*.xcdatamodel`) and `PBXVariantGroup` (like `*.strings`)
+            let fileRef = try! self.pbxproj.object(id: buildFile.fileRef) as PBXGroup
+            return resolveURL(objectUUID: buildFile.fileRef, object: fileRef)
+        }
+    }
+
+    /// Finds the full path / URL of a PBXReference (`PBXFileReference` of `PBXGroup`) based on the groups it belongs to and their `sourceTree` attribute
+    private func resolveURL<T: PBXReference>(objectUUID: ObjectUUID, object: T) -> URL {
+        if objectUUID == self.pbxproj.rootProject.mainGroup { return URL(fileURLWithPath: ".", relativeTo: projectDirectory) }
+
+        switch object.sourceTree {
+        case .absolute:
+            return URL(fileURLWithPath: object.path!)
+        case .group:
+            guard let parentUUID = referrers[objectUUID] else { fatalError("Unable to find parent of \(object) (\(objectUUID))") }
+            let parentGroup = try! self.pbxproj.object(id: parentUUID) as PBXGroup
+            let groupURL = resolveURL(objectUUID: parentUUID, object: parentGroup)
+            return object.path.map { groupURL.appendingPathComponent($0) } ?? groupURL
+        case .projectRoot:
+            return object.path.map { URL(fileURLWithPath: $0, relativeTo: projectDirectory) } ?? projectDirectory
+        case .buildProductsDir, .devDir, .sdkDir:
+            fatalError("Unsupported relative reference (relative to: \(object.sourceTree)")
```

Comment:
We don't (and shouldn't) have files in our pbxproj referenced relative to SDK or BuildProducts or Dev Dir, so that's not a case we will encounter nor need to implement right now…

But maybe I should make this throw instead, and just print a warning + skip any file found in the pbxproj that might end up using this in the future?